website/blog/2020-02-06-react-navigation-5.0.md

```diff
@@ -0,0 +1,145 @@
+
+Finally, we're proud to announce the stable version of React Navigation 5.
+
+## Highlights
+
+### Component based configuration
+
+In previous versions of React Navigation, we used configure the navigator statically using `createXNavigator` functions and `static navigationOptions`. In React Navigation 5, all of the configuration happens inside a component and is dynamic.
+
+Example:
+
+```js
+function App() {
+  return (
+    <Stack.Navigator initialRouteName="home">
+      <Stack.Screen name="settings" component={Settings} />
+      <Stack.Screen
+        name="profile"
+        component={Profile}
+        options={{ title: 'John Doe' }}
+      />
+    </Stack.Navigator>
+  );
+}
+```
+
+This means we have access to props, state and context, and can dynamically change the configuration for the navigator!
+
+### New hooks
+
+Hooks are great for stateful logic and code organization. Now we have several hooks for common use cases:
+
+- [`useNavigation`](https://reactnavigation.org/docs/use-navigation.html)
+- [`useRoute`](https://reactnavigation.org/docs/use-route.html)
+- [`useNavigationState`](https://reactnavigation.org/docs/use-navigation-state.html)
+- [`useFocusEffect`](https://reactnavigation.org/docs/use-focus-effect.html)
+- [`useIsFocused`](https://reactnavigation.org/docs/use-is-focused.html)
+- [`useLinking`](https://reactnavigation.org/docs/use-linking.html)
+- [`useScrollToTop`](https://reactnavigation.org/docs/use-scroll-to-top.html)
+
+### Update options from component
+
+We’ve added a new [`setOptions`](https://reactnavigation.org/docs/navigation-prop.html#setoptions---update-screen-options-from-the-component) method on the `navigation` prop to make configuring screen navigation options more intuitive than its `static navigationOptions` predecessor. It lets us **easily set screen options based on props, state or context without messing with params**. Instead of using static options, we can call it anytime to configure the screen.
+
+```js
+navigation.setOptions({
+  headerRight: () => (
+    <DoneButton
+      onPress={async () => {
+        await saveNote();
+        navigation.replace('Notes');
+      }}
+    />
+  ),
+})
+```
+
+It can be used fo things like adding a button in the header which needs to interact with the screen state.
+
+### New theming API
+
+In React Navigation, we had basic theming support where you could specify whether to use a light or dark theme. It wasn't easy to customize the colors used by the built-in components such as header, tab bar etc. without extra code or repetition.
+
+Now, we have revamped the [theme system](https://reactnavigation.org/docs/themes.html) for easier customization. Its possible to provide a theme object with your desired colors for background, accent color etc. and it will automatically change the colors of all navigators without any extra code.
```

Comment:
```suggestion
Now, we have revamped the [theme system](https://reactnavigation.org/docs/themes.html) for easier customization. It is possible to provide a theme object with your desired colors for background, accent color etc. and it will automatically change the colors of all navigators without any extra code.
```